common.tf

```diff
@@ -9,16 +9,33 @@ locals {
   apiary_bucket_prefix                 = "${local.instance_alias}-${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}"
   apiary_assume_role_bucket_prefix     = [for assumerole in var.apiary_assume_roles : "${local.instance_alias}-${data.aws_caller_identity.current.account_id}-${lookup(assumerole, "allow_cross_region_access", false) ? "*" : data.aws_region.current.name}"]
   enable_route53_records               = "${var.apiary_domain_name == "" ? "0" : "1"}"
-  apiary_managed_schema_names_original = [for schema in var.apiary_managed_schemas : "${schema.schema_name}"]
-  apiary_managed_schema_names_replaced = [for schema in var.apiary_managed_schemas : "${replace(schema.schema_name, "_", "-")}"]
-  apiary_data_buckets                  = [for schema in var.apiary_managed_schemas : "${local.apiary_bucket_prefix}-${replace(schema.schema_name, "_", "-")}"]
+  #
+  # Create a new list of maps with some extra attributes needed later
+  #
+  schemas_info = [
+    for schema in var.apiary_managed_schemas: merge(
+      {
+        replaced_name : replace(schema["schema_name"], "_", "-"),
+        data_bucket   : "${local.apiary_bucket_prefix}-${replace(schema["schema_name"], "_", "-")}"
+      },
+      schema)
+  ]
```

Comment:
nice!!, simpler way to manage bucket names.

may be we can rename `replaced_name` as `resource_name/resource_suffix` ...etc as that is what we are using for.

Reply:
can also add in another PR

Reply:
done